packer/builder/azure/utils/BuildAzureVmName.go

```diff
@@ -8,22 +8,52 @@ import (
 	"math/rand"
 	"os"
 	"time"
-	"sync"
 )
 
 const availableSymbols = "0123456789abcdefghijklmnopqrstuvwxyz"
-const allowedVmNameLength = 15
 
-var randmu sync.Mutex
+var pwSymbols = []string{
+	"abcdefghijklmnopqrstuvwxyz",
+	"ABCDEFGHIJKLMNOPQRSTUVWXYZ",
+	"0123456789",
+}
+
+const allowedVmNameLength = 15
 
 func BuildAzureVmNameRandomSuffix(prefix string) (suffix string) {
-	randmu.Lock()
-	rand.Seed(time.Now().UnixNano() + int64(os.Getpid()))
-	availableSymbolsLen := len(availableSymbols)
 	genLen := allowedVmNameLength - len(prefix)
-	for i := 0; i < genLen; i++ {
-		suffix += string(availableSymbols[rand.Intn(availableSymbolsLen)])
+	suffix += RandomString(availableSymbols, genLen)
+	return
+}
+
+var rnd = rand.New(rand.NewSource(time.Now().UnixNano() + int64(os.Getpid())))
```

Comment:
Seed is pid-based. Differerence is that it's only seeded once per process (i.e. static/global).